eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -267,7 +267,7 @@ jobs:
             searchFolder: '$(System.DefaultWorkingDirectory)/sdk'
             testResultsFiles: "**/test-results.xml"
             testRunTitle: "$(OSName) - NodeJS - Unit Tests - [Node $(NodeVersion)]"
-          condition: succeededOrFailed()
```

Comment:
Interesting that succeededOrFailed wasn't enough! I guess cancellation doesn't fall within that binary.

Reply:
Right, there are 3 states (succeeded, failed, cancelled).  Most of the time `succededOrFailed()` is what you want, since if the build was cancelled you don't want to run any more steps.  However, for a "cleanup" step like publishing test results, you do want to run if cancelled.

Reply:
Definitely, thanks for catching that